src/gamestate/items.rs

```diff
@@ -240,6 +240,73 @@ impl Item {
         }
     }
 
+    /// Checks if this item is a weapon.
+    #[must_use]
+    pub const fn is_weapon(&self) -> bool {
+        matches!(self.typ, ItemType::Weapon { .. })
+    }
+
+    /// Checks if a character of the given class can equip this item.
+    ///
+    /// Returns `true` if the item does not have a class requirement, or if the
+    /// class requirement matches the given class.
+    #[must_use]
+    pub fn can_class_equip_this(&self, class: Class) -> bool {
```

Comment:
I think it would make sense for this be a method on Class called `can_equip(&self, item: &Item)`. That way it is shorter and if you want to extend this to something like companions, where the warrior does not have a shield, you just implement a `can_equip` function on Companion, instead of `can_companion_equip_this()`, or `can_figher_equip_this()` on item. 

Might also be worth defining how consumeables (potions, etc.) fit into this, since you would assume that you can not equip a potion, but (as long as I am not missing something here), since they do not have a class requirement, this would return true. Either this should be a more generic ~`can_use`, or this should check if the itemtype has a equipmentslot. Both would be fine by me.